Lib/datetime.py

```diff
@@ -158,26 +169,29 @@ def _build_struct_time(y, m, d, hh, mm, ss, dstflag):
     dnum = _days_before_month(y, m) + d
     return _time.struct_time((y, m, d, hh, mm, ss, wday, dnum, dstflag))
 
-def _format_time(hh, mm, ss, us, timespec='auto'):
+
+def _format_time(hh, mm, ss, us, sus, timespec='auto'):
     specs = {
         'hours': '{:02d}',
         'minutes': '{:02d}:{:02d}',
         'seconds': '{:02d}:{:02d}:{:02d}',
         'milliseconds': '{:02d}:{:02d}:{:02d}.{:03d}',
-        'microseconds': '{:02d}:{:02d}:{:02d}.{:06d}'
+        'microseconds': '{:02d}:{:02d}:{:02d}.{:06d}',
+        'submicroseconds': '{:02d}:{:02d}:{:02d}.{:06d}{:03d}',
     }
 
     if timespec == 'auto':
-        # Skip trailing microseconds when us==0.
-        timespec = 'microseconds' if us else 'seconds'
+        # Skip trailing subseconds when sus==0 or us==0.
+        timespec = 'submicroseconds' if us or sus else 'seconds'
```

Comment:
This changes the default behavior to include both micro and nanoseconds even if the precision doesn't include any nanoseconds, changing the behavior of existing code. There's gotta be a strong reason for making existing code suddenly change behavior.